### DISCUSSION TITLE
difficulties to import databases

### DISCUSSION THREAD

**Discussion by biosupi** [aertslab/SCENIC#271] [R]
Hi, 

Congratulation for this impressive work. I'm not bio-informatician (but biologist), and I would like to use the Scenic pipeline. However, when following the "running scenic" tutorial, I'm blocked at the "initializing Scenic settings" step with the following error : 

Error in initializeScenic(org = org, dbDir = dbDir, dbs = dbs, datasetTitle = myDatasetTitle,  : 
  The following RcisTarget databases were not found: 
- cisTarget_databases/mm9-500bp-upstream-7species.mc9nr.feather 
- cisTarget_databases/mm9-tss-centered-10kb-7species.mc9nr.feather
Make sure the arguments 'dbDir' and 'dbs' are correct.

I think that I've probably a problem with the downloading of the databases, but I did not find the problem by myself (I tried to use the "https://resources-mirror.aertslab.org/cistarget/databases")...

Could you give me a hand for this step ? 

Thanks a lot

**Comment by s-aibar** [comment]
Are you modifying the default arguments? Once you have successfully downloaded the databases, you have to tell the function:
1. In which location (folder) they are saved (indicating it with the argument `dbDir`)
2. The name of the files: this can be done through `organism` (e.g. there are some default databases for each organism, in your case it is using the mouse ones). If you want to use different databases (or rename them), you can provide the new names with `dbs`)

**Comment by MubasherMohammed** [comment]
@s-aibar Thanks for your assistance its great work you guys achieved!! I actually have the same issue I keep getting the same error I have tired many times different ways , so I downloaded the databases manually on the "cis_Targetdatabases" as am following the mouse brain tutorial, can you please advice on how to overcome this..attached screenshot of the error, also, I would like to use the package on Plasmodium falciparum species do you recommed following specific guidelines for this? TIA 
<img width="1249" alt="Screenshot 2020-04-26 at 11 53 40" src="https://user-images.githubusercontent.com/50444056/80304405-d3c5b000-87b5-11ea-937d-8d0e6e4577af.png">
<img width="1249" alt="Screenshot 2020-04-26 at 11 53 40" src="https://user-images.githubusercontent.com/50444056/80304408-db855480-87b5-11ea-92c0-355d3749ed68.png">

**Comment by s-aibar** [comment]
@MubasherMohammed  : 

I suspect that the error might have something to do with using the  `~` symbol to replace the home directory... 

We have just pushed a new version with some minor updates (including ` path.expand` to the databases directory). That could solve your problem. 

If it doesn't, we will need some more information... Could you check whether this code works? 

```
library(SCENIC)
org <- "mgi" # or hgnc, or dmel
dbDir <- "cisTarget_databases" # RcisTarget databases location
dbDir <- path.expand(dbDir)

data(defaultDbNames)
dbs <- defaultDbNames[[org]]
dbs

lapply(dbs,function(x) file.exists(file.path(dbDir, x)))
```

**Comment by fandi-2020** [comment]
> @MubasherMohammed :
> 
> I suspect that the error might have something to do with using the `~` symbol to replace the home directory...
> 
> We have just pushed a new version with some minor updates (including ` path.expand` to the databases directory). That could solve your problem.
> 
> If it doesn't, we will need some more information... Could you check whether this code works?
> 
> ```
> library(SCENIC)
> org <- "mgi" # or hgnc, or dmel
> dbDir <- "cisTarget_databases" # RcisTarget databases location
> dbDir <- path.expand(dbDir)
> 
> data(defaultDbNames)
> dbs <- defaultDbNames[[org]]
> dbs
> 
> lapply(dbs,function(x) file.exists(file.path(dbDir, x)))
> ```

Hi there. I had tried the code you suggested but I still got the error (as below), do you have any suggestions on this ? many thanks!
 
org <- "mgi" 
> dbDir <- "/Users/fd/Desktop/cisTarget_databases/mm9-500bp-upstream-7species.mc9nr.feather"
> dbDir <- path.expand(dbDir)
> data(defaultDbNames)
> dbs <- defaultDbNames[[org]]
> dbs
                                         500bp                                           10kb 
   "mm9-500bp-upstream-7species.mc9nr.feather" "mm9-tss-centered-10kb-7species.mc9nr.feather" 
> lapply(dbs,function(x) file.exists(file.path(dbDir, x)))
$`500bp`
[1] FALSE

$`10kb`
[1] FALSE

**Comment by s-aibar** [comment]
Dear @fandi-2020 , 

According to that it seems that the files are not in the right location... (or that there is an issue finding them...)

Can you check whether they are indeed in the folder that you are providing? 
```
### Check if they exist:
for(x in dbs) { 
  dbPath <- file.path(dbDir, x)
  message(paste(dbPath, " --> Exists?:", file.exists(dbPath))) 
}
```

e.g. In my case:
```
dbDir <- ".../cisTarget_databases" # RcisTarget databases location

### Prepare... 
library(SCENIC)
org <- "mgi" # or hgnc, or dmel
data(defaultDbNames)
dbs <- defaultDbNames[[org]]

### Check if they exist:
for(x in dbs) { 
  dbPath <- file.path(dbDir, x)
  message(paste(dbPath, " --> Exists?:", file.exists(dbPath))) 
}

## Output:
/.../cisTarget_databases/mm9-500bp-upstream-7species.mc9nr.feather  --> Exists?: TRUE
/.../cisTarget_databases/mm9-tss-centered-10kb-7species.mc9nr.feather  --> Exists?: TRUE
```
